utils/s3/credentials_providers/sts_assume_role_credentials_provider.cc

```diff
@@ -57,6 +57,7 @@ future<> sts_assume_role_credentials_provider::update_credentials() {
             creds = parse_creds(body);
         },
         http::reply::status_type::ok);
+    co_await http_client.close();
```

Comment:
It looks like a standalone bug-fix, no? I mean -- even before this PR, the temporary client in sts and instance-profile providers was left unclosed

Reply:
right

Reply:
Then it deserves its own patch